internal/querynode/load_index_info.go

```diff
@@ -149,6 +149,34 @@ func (li *LoadIndexInfo) appendIndexData(bytesIndex [][]byte, indexKeys []string
 		}
 	}
 
-	status = C.AppendIndex(li.cLoadIndexInfo, cBinarySet)
+	// TODO::xige-16 support embedded milvus
+	storageType := "minio"
+	cAddress := C.CString(Params.MinioCfg.Address)
+	cBucketName := C.CString(Params.MinioCfg.BucketName)
+	cAccessKey := C.CString(Params.MinioCfg.AccessKeyID)
+	cAccessValue := C.CString(Params.MinioCfg.SecretAccessKey)
+	cRootPath := C.CString(Params.MinioCfg.RootPath)
+	cStorageType := C.CString(storageType)
+	cIamEndPoint := C.CString(Params.MinioCfg.IAMEndpoint)
+	defer C.free(unsafe.Pointer(cAddress))
+	defer C.free(unsafe.Pointer(cBucketName))
+	defer C.free(unsafe.Pointer(cAccessKey))
+	defer C.free(unsafe.Pointer(cAccessValue))
+	defer C.free(unsafe.Pointer(cRootPath))
+	defer C.free(unsafe.Pointer(cStorageType))
+	defer C.free(unsafe.Pointer(cIamEndPoint))
+	storageConfig := C.CStorageConfig{
+		address:          cAddress,
+		bucket_name:      cBucketName,
+		access_key_id:    cAccessKey,
+		access_key_value: cAccessValue,
+		remote_root_path: cRootPath,
+		storage_type:     cStorageType,
+		iam_endpoint:     cIamEndPoint,
+		useSSL:           C.bool(Params.MinioCfg.UseSSL),
+		useIAM:           C.bool(Params.MinioCfg.UseIAM),
+	}
+
+	status = C.AppendIndex(li.cLoadIndexInfo, cBinarySet, storageConfig)
```

Comment:
suggest move storageConfig to cLoadIndexInfo

Reply:
why?

Reply:
There is no need to expose one more parameter, you can directly specify storageConfig when constructing LoadIndexInfo.
The underlying code just takes the storageConfig from LoadIndexInfo.